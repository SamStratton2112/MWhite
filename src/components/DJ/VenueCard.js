import { Col, Button, Container} from 'react-bootstrap';
import {useState} from 'react';

const VenueCard = ({venue}) => {
    const [hidden, setHidden] = useState(true);

    let body1 = <img src={venue.image}  style={{ width: '19rem', height: '19rem' }} alt="coming soon" />;
    let body2 = <a href={venue.src}><Button>Visit {venue.name}</Button></a>;
    return (
        <Col>
            <Container 
                style={{ width: '20rem', height: '20rem', border: 'solid' }} 
                className="venueCard" 
                onMouseEnter={() => setHidden(false)} 
                onMouseLeave={() => setHidden(true)}
            >
                {hidden ? body1 : body2}
            </Container>
        </Col>
    );
};

export default VenueCard;
