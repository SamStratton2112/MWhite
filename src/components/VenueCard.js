import { Col, Container} from 'react-bootstrap';
import {useState} from 'react';

/** Component displays residencies images except when user hovers over the image which is replaced by a button linking to the venues website.*/

const VenueCard = ({venue}) => {
    const [hidden, setHidden] = useState(true);

    let body1 = <img src={venue.image}  style={{ width: '19rem', height: '19rem', borderRadius: '.2em',  border:'solid', borderWidth: '.2em',
    borderColor: 'rgb(184, 17, 17)' }} alt={venue.name} />;

    let body2 = <a href={venue.src}><button className="bttn">Visit {venue.name}</button></a>;
    return (
        <Col>
            <Container 
                style={{ width: '20rem', height: '20rem' }} 
                className="venueCard center" 
                onMouseEnter={() => setHidden(false)} 
                onMouseLeave={() => setHidden(true)}
            >
                {hidden ? body1 : body2}
            </Container>
        </Col>
    );
};

export default VenueCard;
