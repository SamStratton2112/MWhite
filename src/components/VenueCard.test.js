import VenueCard from './VenueCard';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

const testVenue = {
    image: 'testImage',
    name: 'testName'
}

describe(VenueCard, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <VenueCard venue={testVenue}/>
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <VenueCard venue={testVenue}/>
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});