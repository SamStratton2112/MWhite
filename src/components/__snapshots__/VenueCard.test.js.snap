// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenueCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <div
      class="venueCard center container"
      style="width: 20rem; height: 20rem;"
    >
      <img
        alt="testName"
        src="testImage"
        style="width: 19rem; height: 19rem; border-radius: .2em; border: solid; border-width: .2em; border-color: rgb(184, 17, 17);"
      />
    </div>
  </div>
</DocumentFragment>
`;
